internal/rpc/streamproxy_test.go

```diff
@@ -84,7 +84,32 @@ func TestStreamProxyStoppedController(t *testing.T) {
 	ctx := context.Background()
 
 	doneChan := make(chan error)
-	stopped := false
+	srvController := newServer(func(c *websocket.Conn) error { return errors.New("stopped") })
+	srvJIMM := newServer(func(connClient *websocket.Conn) error {
+		connController, err := srvController.dialer.DialWebsocket(ctx, srvController.URL, nil)
+		c.Assert(err, qt.IsNil)
+		rpc.ProxyStreams(ctx, connClient, connController)
+		doneChan <- nil
+		return nil
+	})
+	defer srvController.Close()
+	defer srvJIMM.Close()
+	ws, err := srvJIMM.dialer.DialWebsocket(ctx, srvJIMM.URL, nil)
+	c.Assert(err, qt.IsNil)
+	defer ws.Close()
+
+	verifyEcho(c, ws, ".*abnormal closure.*")
+
+	ws.Close()
+	<-doneChan // Ensure go routines are cleaned up
+}
+
+func TestStreamProxyStoppedMidwayController(t *testing.T) {
```

Comment:
I think this test is okay, but it doesn't need the `stopped` variable. I tested it locally and commented out the line `stopped.Store(true)` and it still passes because the streamEcho returns after echoing 1 message.

Reply:
even simpler